app/views/applications/new.html.erb

```diff
@@ -1,8 +1,8 @@
 <h1>New Application Form</h1>
 <%= form_tag("/applications") do %>
 <% @pets.each do |pet| %>
-  <p><%= check_box_tag "pet_id-#{pet.id}", "#{pet.id}" %>
-  <%= pet.name %></p>
+  <p id="checkbox-<%= pet.id %>"><%= check_box_tag "applied_for[]", pet.id %>
```

Comment:
All ids for checkbox fields will be the same this way - the unique ids are moved to the paragraphs the checkbox is contained in